doc/releases/changelog-dev.md

```diff
@@ -16,6 +16,10 @@
 
 <h3>Improvements</h3>
 
+* The `num_params` attribute in the operator class is now dynamic. This makes it easier
+  to define operator subclasses with a flexible number of parameters. 
+  [(#1898)](https://github.com/PennyLaneAI/pennylane/pull/1898)
```

Comment:
I haven't finished going through the PR yet, but does this need to be added to breaking changes as well?

Reply:
I moved it! I am still so confused, in some sense every change (but docstring changes) is a breaking change? But you are right, this is kind of user facing since it is not private.